Source/WebCore/rendering/RenderFlexibleBox.cpp

```diff
@@ -1709,6 +1706,9 @@ FlexItem RenderFlexibleBox::constructFlexItem(RenderBox& child, bool relayoutChi
     if (childHadLayout && child.hasTrimmedMargin(std::optional<MarginTrimType> { }))
         child.clearTrimmedMarginsMarkings();
     
+    if (child.needsPreferredWidthsRecalculation())
+        child.setPreferredLogicalWidthsDirty(true, MarkingBehavior::MarkOnlyThis);
+
```

Comment:
I think it'd be better to have a dedicated function to run invalidation on flex items in a similar way we do it in block layout, see `RenderBlock::updateBlockChildDirtyBitsBeforeLayout` (as opposed to put it in a seemingly random function)

Reply:
sure.